website/blog/deployers.md

```diff
@@ -0,0 +1,592 @@
+# How to Implement a Service Weaver Deployer
+
+<div class="blog-author">Michael Whittaker</div>
+<div class="blog-date">TODO, 2023</div>
+
+Service Weaver allows you to deploy an application binary in many different
+ways. For example, you can deploy an application in a [single
+process][singleprocess], across [multiple processes][multiprocess], or in [the
+cloud][gke]. The code that deploys a Service Weaver application is called,
+unsurprisingly, a **deployer**. In this blog post, we provide a high-level
+overview of *what* deployers are and then get down to the nitty-gritty of
+exactly *how* they work, culminating in us implementing one completely from
+scratch. This blog post assumes you have a basic familiarity with how to write
+Service Weaver applications. If you don't, we recommend you read the
+[step-by-step tutorial][tutorial].
+
+## Weavelets
```

Comment:
I would structure this a little different:

Start with a picture, that visually show everything at high-level, the app, the components,the weavelet, the deployer, the envelopes. Then I will introduce weavelet, deployer (and envelope as part of the deployer).
 This might be it's own section, maybe.

This will allow someone to quickly keep track of the main things related to runtime and app/runtime interactions, and get a sense where different parts you are gonna text next fit in the picture.

Then the next section will be, okay, now we learn about the concepts and have an idea how they relate to each other, let's actually write deployers, and then present the multiprocess deployer, and the appendix.

Reply:
Other comments:
I would make envelopes as part of the deployers section.

For me it reads better if we have:

Deployers:
Main responsibilitlies

Envelopes